cookbook/case_studies/house_price_prediction/house_price_predictor.py

```diff
@@ -0,0 +1,232 @@
+def gen_price(house) -> int:
+    _base_price = int(house["SQUARE_FEET"] * 150)
+    _price = int(
+        _base_price
+        + (10000 * house["NUM_BEDROOMS"])
+        + (15000 * house["NUM_BATHROOMS"])
+        + (15000 * house["LOT_ACRES"])
+        + (15000 * house["GARAGE_SPACES"])
+        - (5000 * (MAX_YEAR - house["YEAR_BUILT"]))
+    )
+    return _price
+
+
+# %%
+# Dataframe with All the House Details:
+def gen_houses(num_houses) -> pd.DataFrame:
+    _house_list = []
+    for _ in range(num_houses):
+        _house = {
+            "SQUARE_FEET": int(np.random.normal(3000, 750)),
+            "NUM_BEDROOMS": np.random.randint(2, 7),
+            "NUM_BATHROOMS": np.random.randint(2, 7) / 2,
+            "LOT_ACRES": round(np.random.normal(1.0, 0.25), 2),
+            "GARAGE_SPACES": np.random.randint(0, 4),
+            "YEAR_BUILT": min(MAX_YEAR, int(np.random.normal(1995, 10))),
+        }
+        _price = gen_price(_house)
+        _house_list.append(
+            [
+                _price,
+                _house["YEAR_BUILT"],
+                _house["SQUARE_FEET"],
+                _house["NUM_BEDROOMS"],
+                _house["NUM_BATHROOMS"],
+                _house["LOT_ACRES"],
+                _house["GARAGE_SPACES"],
+            ]
+        )
+    _df = pd.DataFrame(
+        _house_list,
+        columns=COLUMNS,
+    )
+    return _df
+
+
+# %%
+# Splitting the Data into Train, Val, and Test Datasets:
+def split_data(
+    df: pd.DataFrame, seed: int, split: typing.List[float]
+) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+
+    seed = seed
+    val_size = split[1]
+    test_size = split[2]
+
+    num_samples = df.shape[0]
+    x1 = df.values[
+        :num_samples, 1:
+    ]  # keep only the features, skip the target, all rows
+    y1 = df.values[:num_samples, :1]  # keep only the target, all rows
+
+    # Use split ratios to divide up into train & test
+    x_train, x_test, y_train, y_test = train_test_split(
+        x1, y1, test_size=test_size, random_state=seed
+    )
+    # Of the remaining training samples, give proper ratio to train & validation
+    x_train, x_val, y_train, y_val = train_test_split(
+        x_train,
+        y_train,
+        test_size=(val_size / (1 - test_size)),
+        random_state=seed,
+    )
+
+    # Reassemble the datasets with target in first column and features after that
+    _train = np.concatenate([y_train, x_train], axis=1)
+    _val = np.concatenate([y_val, x_val], axis=1)
+    _test = np.concatenate([y_test, x_test], axis=1)
+
+    return (
+        pd.DataFrame(
+            _train,
+            columns=COLUMNS,
+        ),
+        pd.DataFrame(
+            _val,
+            columns=COLUMNS,
+        ),
+        pd.DataFrame(
+            _test,
+            columns=COLUMNS,
+        ),
+    )
+
+
+# %%
+# Step 4: Task -- Generating & Splitting the Data
+# -----------------------------------------------
+#
+# Train, validation, and test datasets are DataFrames.
```

Comment:
lets explain why we generated and split the data in one task?

Reply:
I don't think there's a specific reason as to why that's done.

Reply:
It's just that there are two helper functions that we've already defined, and hence, we are calling them in a single function. Do you want me to write this?